### PR TITLE
Edit Identity Trait [Milestone 1]

### LINKED ISSUE #18: Edit Identity Trait
add extrinsic for editing identity traits

### PR DESCRIPTION
**Pull request by isaacadams**
added edit trait identity function

Closes #18

### CODE REVIEW

File: pallets/identity/src/lib.rs [hunk 1]
```diff
@@ -127,7 +127,28 @@ pub mod pallet {
...
+        }
+
+        #[pallet::weight(10_000 + T::DbWeight::get().writes(1))]
+        /// Edit an existing identity trait
+        pub fn edit_identity_trait(origin: OriginFor<T>, identity_id: u32, key: Vec<u8>, value: Vec<u8>) -> DispatchResult {
+            let who = ensure_signed(origin)?;
+
+            ensure!(Self::is_identity_owned_by_sender(&who, &identity_id), Error::<T>::IdentityNotOwned);
+
+            <IdentityTraitList<T>>::try_mutate(identity_id, key, |v| -> DispatchResult {
```
Comment (Romulus10) [inline pallets/identity/src/lib.rs:147]: Are these two functions identical?

Comment (isaacadams) [reply pallets/identity/src/lib.rs:147]: yes -- that is why I was suggesting we make it one function and call it `upsert`

Comment (isaacadams) [reply pallets/identity/src/lib.rs:147]: `try_mutate` makes it possible to handle both insert and update using the same code.

Comment (Romulus10) [reply pallets/identity/src/lib.rs:147]: Yep that makes more sense now. I knew the `try_mutate` section would be the same but I didn't realize fully the extent to which the functions were legitimately just the same thing. I do think `upsert` is a bit of an unclear name - maybe `modify`?

Comment (isaacadams) [reply pallets/identity/src/lib.rs:147]: `modify` delivers the concept of editing, or updating -- but I don't feel like it delivers the idea of "add" or "insert".

I am not aware of another term typically used to denote the action of both an "update" and an "insert" other than "upsert"

Comment (isaacadams) [reply pallets/identity/src/lib.rs:147]: we could also mirror the internal method we are using, that is `mutate_identity_trait`

Comment (Romulus10) [reply pallets/identity/src/lib.rs:147]: Yeah, fair point well made. Modify isn't really sufficient, is it. I usually just use "insert_or_update", "add_or_update", my main hesitation is I've never actually seen "upsert" used that I can remember.

Comment (Romulus10) [reply pallets/identity/src/lib.rs:147]: > we could also mirror the internal method we are using, that is `mutate_identity_trait`

Yeah mutate had occurred to me too, and I actually had the same concern as you brought up with `modify`.

**Review by Romulus10 — APPROVED** [review]

**Pull request merged**